pkg/controller/controllerutil/controllerutil.go

```diff
@@ -181,6 +181,18 @@ func HasControllerReference(object metav1.Object) bool {
 	return false
 }
 
+// HasOwnerReference returns true if the owners list contains an owner reference
+// that matches the object's group, kind, and name.
+func HasOwnerReference(owners []metav1.OwnerReference, obj client.Object) bool {
+	gvk := obj.GetObjectKind().GroupVersionKind()
```

Comment:
There is a function `apiutil.GVKForObject()` that would be useful here.  You would have to interface assert this to a runtime object.

Reply:
+1, `TypeMeta` may not always exist in the object, even c-r has made some workaround for it
https://github.com/kubernetes/kubernetes/issues/3030
https://github.com/kubernetes/kubernetes/issues/80609

Reply:
I rewrote like [this](https://github.com/kubernetes-sigs/controller-runtime/pull/2882/commits/364384a51c8a2ee2c9830b850cf622d7d8d7f0cb).
Is it right, I want the feedback, thanks.